core/lib/zksync_core/src/lib.rs

```diff
@@ -828,8 +828,11 @@ async fn add_state_keeper_to_task_futures(
     batch_fee_input_provider: Arc<dyn BatchFeeModelInputProvider>,
     stop_receiver: watch::Receiver<bool>,
 ) -> anyhow::Result<()> {
-    let pool_builder = ConnectionPool::<Core>::singleton(postgres_config.master_url()?);
-    let state_keeper_pool = pool_builder
+    // One (potentially held long-term) connection for async catch up task and another connection for
+    // batch executor.
+    let state_keeper_pool_builder =
+        ConnectionPool::<Core>::builder(postgres_config.master_url()?, 2);
```

Comment:
IMO, it would be better if we had 3 singleton pools for:
1. AsyncCatchupTask
2. AsyncRocksdbCache
3. MempoolIO

or at least allow 3 connections per shared pool

Reply:
I split this pool into two as @slowli suggested: a two-sized pool for `AsyncCatchupTask/AsyncRocksdbCache ` and a separate singleton pool for MempoolIO